fMRISurface/GenericfMRISurfaceProcessingPipeline_1res.sh

```diff
@@ -62,6 +62,49 @@ fi
 
 RUN=`opts_GetOpt1 "--printcom" $@`  # use ="echo" for just printing everything and not running the commands (default is to run)
 
+# ------------------------------------------------------------------------------
+#  Check MMP Version
+# ------------------------------------------------------------------------------
+
+Compliance="hcp"
+MPPVersion=`opts_GetOpt1 "--mppversion" $@`
+MPPVersion=`opts_DefaultOpt $MPPVersion "hcp"`
+
+if [ "${MPPVersion}" = "legacy" ] ; then
+  log_Msg "Legacy Minimal Preprocessing Pipelines: fMRISurface v.XX"
```

Comment:
These checks/messages that are pasted into a bunch of places seem like maybe they should be in a global script or function instead (you could pass in the tool name as a string).

I don't think we really want to start putting hardcoded version numbers in each script, we are unlikely to keep them up to date.  We do currently have a function to show the version of the entire pipelines repo, though I'm not thrilled with it either (because it requires releases to do 2 additional commits: set version to "v4.1.0" or whatever, commit, tag, set version to "Post-v4.1.0", commit).  Unfortunately, since downloading .zip files from github instead of cloning is something we support, I haven't thought of a better solution.

Reply:
Ok. We can put that as a function in a global script and read the version of the pipelines there. What is the function that prints the version?

Reply:
https://github.com/Washington-University/HCPpipelines/blob/master/global/scripts/version.shlib

See:
https://github.com/Washington-University/HCPpipelines/blob/f00cdc5d80d3434640a4aeba929824697b5bcd9c/DiffusionPreprocessing/DiffPreprocPipeline_PreEddy.sh#L71
https://github.com/Washington-University/HCPpipelines/blob/f00cdc5d80d3434640a4aeba929824697b5bcd9c/DiffusionPreprocessing/DiffPreprocPipeline_PreEddy.sh#L190

Though I advocate always using `"$@"` instead of `$@`.  Some other parts of the code just `cat` the version.txt file, and there is also a show_version script...it is a bit of a mess.

Reply:
Actually, to avoid saying "HCP" in legacy mode, you should probably just `cat "$HCPPIPEDIR"/version.txt` into a variable.

Reply:
Added global mppmodecheck.shlib with version checking.